fluentd-gcp-scaler/scaler.sh

```diff
@@ -80,6 +80,39 @@ build_flags() {
   add_limit memory ${MEMORY_LIMIT}
 }
 
+# $1 is {limits,requests}.{cpu,memory}
+# $2 is the desired value
+needs_update() {
+  CURRENT=$(kubectl get ds -n kube-system ${DS_NAME} -o \
+    jsonpath="{.spec.template.spec.containers[?(@.name=='fluentd-gcp')].resources.$1}")
+  DESIRED=$2
+  if [ "${CURRENT}" != "${DESIRED}" ]
+  then
+    log "$1 needs updating. Is: '${CURRENT}', want: '${DESIRED}'."
+    return 0
+  fi
+  return 1
+}
+
+update_if_needed() {
+  NEED_UPDATE=false
+  if needs_update requests.cpu ${CPU_REQUEST}; then NEED_UPDATE=true; fi
+  if needs_update requests.memory ${MEMORY_REQUEST}; then NEED_UPDATE=true; fi
+  if needs_update limits.cpu ${CPU_LIMIT}; then NEED_UPDATE=true; fi
+  if needs_update limits.memory ${MEMORY_LIMIT}; then NEED_UPDATE=true; fi
+  if ! ${NEED_UPDATE}
+  then
+    log "Nothing to update."
+    return
+  fi
+  if [ ${REQUESTS_FLAG} ] || [ ${LIMITS_FLAG} ]
```

Comment:
Is it possible that code reaches this point and $REQUESTS_FLAG and/or $LIMIT_FLAG are empty?

Reply:
Yes, but only under quite bizarre circumstances - when all values are not set anywhere (neither in defaults, nor in scaling policy), but resources are configured in the daemonset. The code above will figure it should update something, but both flags will be empty. Shouldn't happen in practice.